### DISCUSSION TITLE
Pinecil V2 negotiating lower PD power

### DISCUSSION THREAD

**Discussion by colonel-lp** [Ralim/IronOS#1907] [Q&A]
I'm very new to this so would appreciate any advice. I've just got a Pinecil V2 and a miniware ts101 and both are running the latest 2.22 firmware with the same settings. Out of interest I tried plugging them into two identical but separate usb C pd charging ports in my van. The ts101 will negotiate 12.7v but the Pinecil V2 only 9.2v. tried swapping cables and chargers but no difference. Tried maxing out the PD timeout and still no difference. Any ideas?
![20240506_134336](https://github.com/Ralim/IronOS/assets/3628298/9a68b2ed-da96-470a-981a-9fe341d3aefa)
![20240506_134347](https://github.com/Ralim/IronOS/assets/3628298/bd6a34af-5976-450f-a20a-f6d4fa2e1695)

**Comment by ia** [comment]
Hello.

> Tried maxing out the PD timeout and still no difference.

Tried maxing out the PD timeout to do what exactly?

> Any ideas?

I'm not an electrical engineer, but here is my few assumptions with a little rant, if I may.

First of all, according to the provided photos both of the irons are in idle/sleep mode. Therefore, they don't need much of volts since the tips are cooling off anyway.

Second, voltage usage/change based on Ohm's law from the point of physics (i.e., _resistance on a tip_ as the easiest reference) and on _power formula_ (_*wrongfully*_ called _"Watt's law"_ sometimes) _P = I x V_ [programmatically](https://github.com/Ralim/IronOS/blob/ad15a53cd96c6eec12300bcf3ed80bbf0dbdbb04/source/Core/Src/power.cpp#L39).

Therefore (at least that's what _my personal experience shows_ with _TS80P_), once you increase temperature (especially if it's way above _~300 C_) you see right away how voltage will be... okay, _may be_ increased because it depends on not only what's written on the label of a charger or a charging port but on actual _negotiation_ between iron & port (just to illustrate, here is my real case: iron is on _9 V_ or may be even on _5 V_ in idle, but once you need to bring the heat to the tip, it goes right to _12 V_ with _30 W_). And yes, the same chargers for different models of irons may give different results.

Third, keep in mind that these voltage readings are not coming from one of those fancy _TrueRMS_ ®-based _Fluke_ © multimeters but from really cheap chips (_IronOS_ makes miracles even with such cheap & unreliable hardware though!) so I wouldn't trust those readings 100% although I use these readings myself every time as very convenient reference just to get basic idea from the point of understanding _"ok, what's going on with iron power now in the moment?"_.

Now, if your problem is that when you turn on your _Pinecil V2_ and set _350-360 C_ but it doesn't heat properly (i.e. you did measurements for the temperature on the tip) _regardless_ how many volts you see (since lack of volts can be compensated by amps on the wire, for example) <<< now that is _*the problem*_. But then it should be formulated slightly differently in a little bit more clear & more detailed way for more clear understanding of your situation by those who can help you.

So... could you, please, elaborate, what kind of issue are you trying to solve, once again? Do you have enough of heat to solder when your _Pinecil V2_ is up & running & set to _~350 C_?

P.S. If I'm wrong with some of my statements I will be very appreciated for being corrected by way much more knowledgeable people in this field than me. :pray:

**Comment by Ralim** [comment]
Hello, the reason for 9V vs 12V is relatively "simple".

1) The Iron looks at what tip is installed and looks up its resistance (6.2 ohms for Pinecil V2, 8 ohms for TS101 stock tip)
2) The Iron picks the highest USB-PD voltage it can, while ensuring when the tip is turned on we wont draw more current than the supply can give.

Given ohms law of Volts = Current * Resistance; re-arrange this to Current = Volts/Resistance.

Pinecilv2's custom made tips are 6.2 ohms.
TS101's legacy tips are 8 ohms.

PinecilV2 @ 9V = 1.45A = 13W
PinecilV2 @ 12V = 1.9A = 23W

TS101 @ 9V = 1.1A = 10W
TS101 @ 12V = 1.5A = 18W

I would speculate your USB-C charger unit is only advertising a 12V@1.5A profile. So to prevent damage to it, Pinecilv2 is using 9V instead.

If you power up the PinecilV2 while holding the front button; you can see the PD-debugger that will print out what profiles the charger is advertising. (Use buttons to scroll through the information)

**Reply by Ralim** [reply]
We specifically made the PinecilV2 tip to optimise performance for 20V USB-C as that is the most commonly used source.
